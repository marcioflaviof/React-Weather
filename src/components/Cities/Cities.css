div .cities {
  margin-left: 7rem;
  margin-right: 7rem;
}

div .cities .cities-subtitles {
  display: flex;
}

div .cities-subtitles p {
  padding-left: 1rem;
}

div .cities-citie-holder {
  display: flex;
}

div .cities-temp-holder {
  display: flex;
}

div .cities-citie-holder p {
  padding-left: 1.25rem;
  font-weight: bold;
}

div .cities-block-holder {
  display: flex;
}

div .citie-block-one {
  flex: 1;
}
