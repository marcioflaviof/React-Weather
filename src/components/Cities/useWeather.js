import { useEffect, useState } from "react";

const citiesObject = [
  { name: "Rio de Janeiro", min: "17º", max: "23º" },
  { name: "São Paulo", min: "14º", max: "22º" },
  { name: "Belo Horizonte", min: "21º", max: "32º" },
  { name: "Brasília", min: "24º", max: "37º" },
  { name: "Belém", min: "24º", max: "37º" },
  { name: "Salvador", min: "23º", max: "37º" },
  { name: "Curitiba", min: "5º", max: "14º" },
  { name: "Fortaleza", min: "21º", max: "32º" },
  { name: "Manaus", min: "24º", max: "37º" },
  { name: "João Pessoa", min: "28º", max: "40º" },
];

function useWeather() {
  const [cities, setCities] = useState(null);

  useEffect(() => {
    //TODO fetch with yahoo
    const fetchData = () => {
      setCities(citiesObject);
    };

    fetchData();
  });

  console.log(cities);

  return [cities];
}

export default useWeather;
