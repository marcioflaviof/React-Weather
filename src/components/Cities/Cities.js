import { Box, Text } from "@theme-ui/components";
import useWeather from "../../hooks/useWeather";

const sx = {
  column: {
    columns: [null, null, "100px 2"],
    columnGap: "40px",
  },
  span: {
    title: {
      display: "inline-block",
      m: "0 5px",
    },

    city: {
      display: "inline-block",
      minWidth: "18px",
      m: "10px 10px",
    },
  },
};

function Cities() {
  const [cities] = useWeather();

  console.log(cities);

  return (
    <Box
      sx={{
        marginLeft: ["2rem", "2rem", "7.5rem"],
        marginRight: ["2rem", "2rem", "7.5rem"],
      }}
    >
      <Text variant="title">Capitais</Text>

      <Box sx={{ ...sx.column, m: "10px 0" }}>
        <Box>
          <Text variant="small" sx={sx.span.title}>
            Min
          </Text>
          <Text variant="small" sx={sx.span.title}>
            Máx
          </Text>
        </Box>
        <Box
          sx={{
            display: ["none", "none", "inline-block"],
          }}
        >
          <Text variant="small" sx={sx.span.title}>
            Min
          </Text>
          <Text variant="small" sx={sx.span.title}>
            Máx
          </Text>
        </Box>
      </Box>
      <Box sx={sx.column}>
        {cities.length === 10
          ? cities.map((city, index) => (
              <Box key={index}>
                <Text variant="small" sx={sx.span.city}>
                  {city.temperature}
                </Text>
                <Text variant="small" sx={sx.span.city}>
                  {city.name}
                </Text>
              </Box>
            ))
          : "Loading"}
      </Box>
    </Box>
  );
}

export default Cities;
