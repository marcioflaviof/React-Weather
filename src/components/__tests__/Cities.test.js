import React from "react";
import renderer from "react-test-renderer";
import Cities from "../Cities/Cities";

it("renders correctly", () => {
  const cities = renderer.create(<Cities />).toJSON();

  expect(cities).toMatchSnapshot();
});
