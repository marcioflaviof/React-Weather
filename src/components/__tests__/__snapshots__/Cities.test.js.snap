// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-copng3"
>
  <span
    className="css-vurnku"
  >
    Capitais
  </span>
  <div
    className="css-1ksewea"
  >
    <div
      className="css-vurnku"
    >
      <span
        className="css-1ij4mp1"
      >
        Min
      </span>
      <span
        className="css-1ij4mp1"
      >
        Máx
      </span>
    </div>
    <div
      className="css-18bjykf"
    >
      <span
        className="css-1ij4mp1"
      >
        Min
      </span>
      <span
        className="css-1ij4mp1"
      >
        Máx
      </span>
    </div>
  </div>
  <div
    className="css-124fqsq"
  >
    Loading
  </div>
</div>
`;
